tests/kernel_bundle/sycl_compile.h

```diff
@@ -0,0 +1,161 @@
+
+class TestCaseDescription
+    : public kernel_bundle::TestCaseDescriptionBase<sycl::bundle_state::input> {
+ public:
+  TestCaseDescription(std::string_view functionOverload)
+      : TestCaseDescriptionBase<sycl::bundle_state::input>("sycl::compile",
+                                                           functionOverload){};
+};
+
+/** @brief Used to select sycl::compile overload
+ */
+enum class CompileOverload { bundle_and_devs, bundle_only };
+
+using input_bundle_t = sycl::kernel_bundle<sycl::bundle_state::input>;
+
+/** @brief Call sycl::compile overload selected by Overload tparam
+ *  @tparam Overload Selected overload of sycl::compile
+ */
+template <CompileOverload Overload>
+static auto compile_bundle(input_bundle_t &inKb,
+                           const std::vector<sycl::device> &devices) {
+  if constexpr (Overload == CompileOverload::bundle_and_devs) {
+    return sycl::compile(inKb, devices);
+  } else if constexpr (Overload == CompileOverload::bundle_only) {
+    return sycl::compile(inKb);
+  } else {
+    static_assert(Overload != Overload, "Wrong Overload passed");
+  }
+}
+
+/** @brief Check that sycl::compile result bundle contains kernel given
+ *  @tparam DescriptorT Selected kernel descriptor
+ *  @tparam Overload Selected overload of sycl::compile
+ */
+template <typename DescriptorT, CompileOverload Overload>
+static void check_bundle_kernels(util::logger &log, const std::string &kName) {
+  using kernel_t = typename DescriptorT::type;
+
+  auto queue = util::get_cts_object::queue();
+  auto ctx = queue.get_context();
+  auto dev = queue.get_device();
+
+  auto k_id = sycl::get_kernel_id<kernel_t>();
+  // Test can be skipped if all devices do not support online compilation
+  // since this is not sycl::compile's fault
+  if (!sycl::has_kernel_bundle<sycl::bundle_state::input>(ctx, {dev}, {k_id})) {
+    log.skip("No kernel bundle with input state with kernel: " + kName +
+             " (skipped).");
+    return;
+  }
+
+  auto input_kb =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(ctx, {dev}, {k_id});
+  auto input_ids = input_kb.get_kernel_ids();
+
+  auto obj_kb = compile_bundle<Overload>(input_kb, {dev});
+  kernel_bundle::define_kernel<DescriptorT, sycl::bundle_state::input>(queue);
+
+  auto restrictions{
+      kernel_bundle::get_restrictions<DescriptorT,
+                                      sycl::bundle_state::input>()};
+  bool dev_is_compat = restrictions.is_compatible(dev);
+
+  if (obj_kb.has_kernel(k_id) != dev_is_compat) {
+    FAIL(log, "Device does not support kernel " + kName +
+                  " but compiled bundle"
+                  " contains it");
+  }
+
+  // Check that result object bundle has the same kernels as input bundle
+  bool same_kernels_in_obj_bundle =
+      std::any_of(input_ids.cbegin(), input_ids.cend(),
+                  [&](auto &in_id) { return obj_kb.has_kernel(in_id); });
+
+  if (!same_kernels_in_obj_bundle) {
```

Comment:
The interest of the algorithm is that you do not need the variable to compute the intermediate result anymore
```suggestion
  if (!std::any_of(input_ids.cbegin(), input_ids.cend(),
                   [&](auto &in_id) { return obj_kb.has_kernel(in_id); })) {
```

Reply:
I've applied this comment [here](https://github.com/KhronosGroup/SYCL-CTS/pull/226/commits/3b85821013e59bdb8af8b5f011b2105b599fd931#diff-d7dd04b2c33a3f23629a9f87a1e1089058ce7e6a7fa093db890c2d2508b4d206L97-L98)